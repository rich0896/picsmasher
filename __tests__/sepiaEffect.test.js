const { SepiaEffect } = require('../effects.js');
const { createTestCanvas } = require('./helpers/testHelpers.js');

describe('SepiaEffect', () => {
    test('should apply sepia effect correctly', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new SepiaEffect({ intensity: 100 });

        const consoleSpy = jest
            .spyOn(console, 'log')
            .mockImplementation(() => {});

        effect.apply(ctx, canvas);

        expect(consoleSpy).toHaveBeenCalledWith('Applying Sepia Effect');
        consoleSpy.mockRestore();

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });
});
