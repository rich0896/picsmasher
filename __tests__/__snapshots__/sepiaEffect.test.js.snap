// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SepiaEffect should apply sepia effect correctly 1`] = `ImageData { width: 200, height: 200, data: [192, 171, 134, 255, 192, 171, 134, 255, 192, 171...], length: 160000 }`;
